plugin/pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -0,0 +1,69 @@
+/*
```

Comment:
I think that what you can separate is all the changes to ServiceAffinity (which means all the changes in "checkServiceAffinity" and the utils and utils_test.go files). This would significantly reduce the size of this PR.
Do you agree?